### ISSUE TITLE
Image in latent space gets shifted during encoding.

### ISSUE THREAD

**Issue by treeform** [kjsman/stable-diffusion-pytorch#8]
I am using a simple red image as input:

![red](https://user-images.githubusercontent.com/125948/218031608-775f0dec-4d18-4e78-bc14-76f79dedd431.png)

```
from stable_diffusion_pytorch import pipeline
from PIL import Image

prompts = ["a photograph of an astronaut riding a horse"]
input_images = [Image.open('red.png')]
images = pipeline.generate(prompts, input_images=input_images)
images[0].save('output.png')
```

But I am getting the input image shifted down 8px,8px and it generates ugly brown border:

![output](https://user-images.githubusercontent.com/125948/218031759-0d9ce175-4dca-451e-ba91-ace0870cde07.png)

I am pretty sure it happens during the `Encode` pass as its already shifter in latent space. Here is custom dumping of latent space to image:

![encodeDecode](https://user-images.githubusercontent.com/125948/218032116-d473f8f3-33bb-4d7b-8c45-221eff9f90d7.png)

Some thing in the Encode pass that is shifting it by a pixel in the latent space. And I can't figure out what.

**Comment by kjsman** [comment]
Wow, nice catch! It turned out that I implemented encoder's padding in different way with the original one. ([at kjsman/stable-diffusion-pytorch](https://github.com/kjsman/stable-diffusion-pytorch/blob/main/stable_diffusion_pytorch/encoder.py#L13-L19), [at CompVis/stable-diffusion](https://github.com/CompVis/stable-diffusion/blob/main/ldm/modules/diffusionmodules/model.py#L74))

The quick fix -- removing pad at downsampling Conv2d layer and implementing pad at forward method (because asymmetric padding is not supported by PyTorch) -- will be pushed soon.

The better fix -- adding nn.ZeroPad2d layer -- requires revision of the weight file. The problem is, I lost the weight conversion script (this might be also the reply for your prior issue). It was in my old laptop, I migrated to new laptop without that script, and I erased my old laptop. Now I think I should rewrite the script soon, but I can't promise ETA...
